### DISCUSSION TITLE
Is there an example with user roles/permissions ?

### DISCUSSION THREAD

**Discussion by matteoruda** [artgris/FileManagerBundle#96] [General]
Hello! First of all, great package, I have started playing with it today, so I'm very new to this package and I have some questions. I would like to use this bundle in my next projects where I have Companies and Users. Each user belongs to one company. What I would like to achive is:
Each company has its own private folder, something like:
'private/CompanyA'
'private/CompanyB' and so on.
Each user can only see their company folder, so I create a custom service like in the docs:

```
 public function getConf($extra = [])
    {
        $folder = $this->defaultRoot . $this->tokenStorage->getToken()->getUser()->getCompany()->getName();
        $fs = new Filesystem();
        if (!$fs->exists($folder)) {
            $fs->mkdir($folder);
        }
        return ['dir' => $folder ];

    }
```

This works like expected, but I was wondering If i can customize even more.
For example each user can have a personal private folder, so the root would still be the same ( /private/CompanyA ) but UserA would be tho only one able to see his own folder ( /private/CompanyA/UserA ).
Also, would be possible to add a check to disable the delete button unless the user has a certain role (like ROLE_ADMIN) ?
Anyway, thank you for the great package

**Comment by artgris** [comment]
Hi, thank you @matteoruda 
I don't think that's possible with "/private/CompanyA" but maybe with a symlink that point to "/private/shared", for exemple. And you create this symlink after `$fs->mkdir($folder);`

For the delete button, you need to override my bundle : https://symfony.com/doc/current/bundles/override.html#templates
